google-cloud-storage/lib/google/cloud/storage/bucket.rb

```diff
@@ -1169,10 +1169,22 @@ def update if_metageneration_match: nil, if_metageneration_not_match: nil
           updater.check_for_changed_labels!
           updater.check_for_mutable_cors!
           updater.check_for_mutable_lifecycle!
-          return if updater.updates.empty?
-          patch_gapi! updater.updates,
-                      if_metageneration_match: if_metageneration_match,
-                      if_metageneration_not_match: if_metageneration_not_match
+
+          attributes = Array(updater.updates)
+          attributes.flatten!
+          return if attributes.empty?
+          ensure_service!
+          update_args = Hash[attributes.map do |attr|
+            [attr, @gapi.send(attr)]
+          end]
+          update_gapi = API::Bucket.new(**update_args)
+          service.update_bucket name,
```

Comment:
In particular, I notice that `patch_gapi!` sets the `@gapi` instance variable here, whereas your code does not. Is that intentional, and why is it different?

Reply:
I missed this. Thanks for spotting it. I added it now.